src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/Repository/MethodologyVersionRepositoryTests.cs

```diff
@@ -854,6 +854,152 @@ await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId)
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.Methodologies.AddAsync(methodology);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var methodologyRepository = new Mock<IMethodologyRepository>(Strict);
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var service = BuildMethodologyVersionRepository(
+                    contentDbContext: contentDbContext);
+
+                var result = await service
+                    .GetLatestPublishedVersionBySlug("slug");
+
+                VerifyAllMocks(methodologyRepository);
+
+                Assert.NotNull(result);
+                Assert.Equal(latestPublishedVersionId, result.Id);
+                Assert.Equal("slug", result.Slug);
+            }
+        }
+
+        [Fact]
+        public async Task GetLatestPublishedVersionBySlug_OwningPublicationSlug()
+        {
+            var latestPublishedVersionId = Guid.NewGuid();
+            var methodology = new Methodology
+            {
+                LatestPublishedVersionId = latestPublishedVersionId,
+                OwningPublicationSlug = "slug",
+                Versions = new List<MethodologyVersion>
+                {
+                    new()
+                    {
+                        AlternativeSlug = "not-like-this",
+                        Version = 1,
+                        PreviousVersionId = latestPublishedVersionId,
+                    },
+                    new()
+                    {
+                        Id = latestPublishedVersionId,
+                        AlternativeSlug = null,
+                        Version = 0,
+                    },
+                },
+            };
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                await contentDbContext.Methodologies.AddAsync(methodology);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            var methodologyRepository = new Mock<IMethodologyRepository>(Strict);
+
+            await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+            {
+                var service = BuildMethodologyVersionRepository(
+                    contentDbContext: contentDbContext);
+
+                var result = await service
+                    .GetLatestPublishedVersionBySlug("slug");
+
+                VerifyAllMocks(methodologyRepository);
+
+                Assert.NotNull(result);
+                Assert.Equal(latestPublishedVersionId, result.Id);
+                Assert.Null(result.AlternativeSlug);
+                // doesn't return result.Methodology, so cannot check result.Slug/result.Methodology.OwningPublicationSlug
```

Comment:
Typically in this case where we get back just a view model or a partially loaded entity as a result, we'd do one more separate contentDbContext block below, like:

```
await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
{
    // retrieve same entity as "result" above direct from the dbContext. 
    // do checks for result.Slug/result.Methodology.OwningPublicationSlug here
}    
```

Reply:
I can add it, but I don't think it's worth it in this instance because `GetLatestPublishedVersionBySlug` doesn't make any DB changes, so we'd just be checking the test data. I'll leave it for now.